common/hash.c

```diff
@@ -35,6 +35,141 @@
 #include <u-boot/sha512.h>
 #include <u-boot/md5.h>
 
+#if CONFIG_IS_ENABLED(MBEDTLS_LIB_CRYPTO)
+
+static int hash_init_sha1(struct hash_algo *algo, void **ctxp)
+{
+	int ret;
+	mbedtls_sha1_context *ctx = malloc(sizeof(mbedtls_sha1_context));
+
+	if (!ctx)
+		return -ENOMEM;
+
+	mbedtls_sha1_init(ctx);
+	ret = mbedtls_sha1_starts(ctx);
```

Comment:
Why don't you use mbedtls_sha1_starts_ret() instead?

Looking at mbedTLS sources mbedtls_sha1_starts() is going to be deprecated?
Same applies to all mbedtls_shaXXX_starts() calls

Reply:
According to "3.0-migration-guide.md", the "_ret" functions are deprecated and it is correct to use the one without "_ret".